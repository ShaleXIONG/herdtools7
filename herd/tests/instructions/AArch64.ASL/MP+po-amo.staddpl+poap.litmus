AArch64 MP+po-amo.staddpl+poap
Orig=PodWR Amo.StAddPL RfeLA PodRRAP Fre
{
0:X1=x; 0:X2=y;
1:X2=y; 1:X1=x;
}
 P0             | P1           ;
 MOV W0,#1      | LDAR W0,[X2] ;
 STR W0,[X1]    | LDR W3,[X1]  ;
 MOV W3,#1      |              ;
 STADDL W3,[X2] |              ;
exists (1:X0=1 /\ 1:X3=0)
