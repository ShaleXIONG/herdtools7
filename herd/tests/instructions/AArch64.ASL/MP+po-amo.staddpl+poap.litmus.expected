Test MP+po-amo.staddpl+poap Allowed
States 3
1:X0=0; 1:X3=0;
1:X0=0; 1:X3=1;
1:X0=1; 1:X3=1;
No
Witnesses
Positive: 0 Negative: 3
Condition exists (1:X0=1 /\ 1:X3=0)
Observation MP+po-amo.staddpl+poap Never 0 3
Hash=04b3999dee21fd954edf636f27a425c0

